src/js/select.js

```diff
@@ -188,28 +206,59 @@ function Menu(wrapperEl, selectEl) {
 Menu.prototype._createMenuEl = function(wrapperEl, selectEl) {
   var menuEl = doc.createElement('div'),
       optionEls = selectEl.children,
-      numOptions = optionEls.length,
+      numOptions = calculateOptions(optionEls),
       selectedPos = 0,
       optionEl,
       itemEl,
-      i;
+      i,
+      j,
+      pos = 0;
 
   menuEl.className = menuClass;
 
   // add options
-  for (i=0; i < numOptions; i++) {
+  for (i=0; i < optionEls.length; i++) {
```

Comment:
It's faster to cache the value of `optionEls.length` to avoid calculating the length of the array on each iteration of the loop